js/widgets/musickeyboard.js

```diff
@@ -2568,4 +2590,22 @@ function MusicKeyboard() {
 
         return cell;
     };
+
+    this._optimizeNotes = function (selectedNotes){
```

Comment:
please add a comment to explain that this is clustering notes by instrument. (maybe clusterNotes would be a better name?)